src/lib/hooks/use-best-quote/utils/issuance/index.ts

```diff
@@ -34,30 +48,74 @@ export async function getEnhancedRedemptionQuote(
         indexTokenAmount
       )
     console.log('componentsUnits:', componentsUnits)
+    // TODO:
+    const outputTokenAmount = BigInt(0)
+
+    const sender = await signer.getAddress()
+
+    const callData = encodeFunctionData({
+      abi: DebtIssuanceModuleV2Abi,
+      functionName: 'redeem',
+      args: [
+        Ethereum2xFlexibleLeverageIndex.address! as Address,
+        indexTokenAmount,
+        sender as Address,
+      ],
+    })
```

Comment:
The quote functions always prepare these tx's, so already encoding the call data here - if the transaction/trade should happen.